src/Promitor.Agents.ResourceDiscovery/Graph/AzureResourceGraph.cs

```diff
@@ -192,7 +192,8 @@ private Exception CreateUnauthorizedException(List<string> targetSubscriptions)
 
                     break;
                 case AuthenticationMode.UserAssignedManagedIdentity:
-                    unauthorizedException = new UnauthorizedException(QueryIdentityId, targetSubscriptions);
+                    var applicationId = string.IsNullOrWhiteSpace(QueryIdentityId) ? "User Assigned Identity" : QueryIdentityId;
```

Comment:
I would update all these error messages to emphasize that it is externally configured:
```suggestion
                    var applicationId = string.IsNullOrWhiteSpace(QueryIdentityId) ? "Externally configured User Assigned Identity" : QueryIdentityId;
```